felix/bpf/bpfmap/bpf_maps.go

```diff
@@ -34,18 +34,25 @@ import (
 	"github.com/projectcalico/calico/felix/bpf/state"
 )
 
+const (
+	V4MapIndex = iota
+	V6MapIndex
+	CommonMapIndex
+)
+
 type Maps struct {
-	IpsetsMap       maps.Map
+	IpsetsMap    maps.Map
+	ArpMap       maps.Map
+	FailsafesMap maps.Map
+	FrontendMap  maps.Map
+	BackendMap   maps.Map
+	AffinityMap  maps.Map
+	RouteMap     maps.Map
+	CtMap        maps.Map
+	SrMsgMap     maps.Map
+	CtNatsMap    maps.Map
+
```

Comment:
Since we have so many nil pointers in this struct, why not to split it further

```suggestion
type IPMaps {
	IpsetsMap    maps.Map
	ArpMap       maps.Map
	FailsafesMap maps.Map
	FrontendMap  maps.Map
	BackendMap   maps.Map
	AffinityMap  maps.Map
	RouteMap     maps.Map
	CtMap        maps.Map
	SrMsgMap     maps.Map
	CtNatsMap    maps.Map
}

type Generic/CommonMaps {
     <all the other maps>
}

type Maps interface {
    AsSlice() []maps.Map
}
```

As a benefit, then the functions like `getCommonBPFMaps(mapsPtr *Maps) []maps.Map` would not need to return the slice, `AsSlice90` would give it to you.

In any case `CreateBPFMaps(ipv6Enabled bool)` could return `CreateBPFMaps(ipv6Enabled bool) (*IPMaps, *IPMaps, *CommonMaps, error) or ([3]Maps, error).

The whole point of introducing the Maps struct was to be able to pass all the sorts of maps in some condensed way. One struct instead of 15. It might also make sense instead of defining `Maps interface`  to define 

```
type Maps struct {
    Common *CommonMaps
    V4             *IPMaps
    V6             *IPMaps
}
```

That would allow you not to change signature of func like `newBPFEndpointManager()` where you could still pass `*bpfmap.Maps`

Also you would not need to define as you would not have an array :)

```
const (
	V4MapIndex = iota
	V6MapIndex
	CommonMapIndex
)
```